sources/EatHook.cpp

```diff
@@ -76,53 +88,69 @@ bool PLH::EatHook::unHook() {
 	return true;
 }
 
-uint32_t* PLH::EatHook::FindEatFunction(const std::string& apiName, const std::wstring& moduleName) {
+uint32_t* PLH::EatHook::FindEatFunction() {
+	if(!m_moduleBase){
+		m_moduleBase = FindModule();
+	}
+
+	if(!m_moduleBase){
+		Log::log("EAT | Failed to module base", ErrorLevel::SEV);
+		return nullptr;
+	}
+
+	return FindEatFunctionInModule();
+}
+
+uint64_t PLH::EatHook::FindModule() {
 #if defined(_WIN64)
 	PEB* peb = (PPEB)__readgsqword(0x60);
 #else
 	PEB* peb = (PPEB)__readfsdword(0x30);
 #endif
 
-	uint32_t* pExportAddress = nullptr;
-	PEB_LDR_DATA* ldr = (PPEB_LDR_DATA)peb->Ldr;
+	auto* ldr = (PPEB_LDR_DATA)peb->Ldr;
+	auto* dte = (LDR_DATA_TABLE_ENTRY*)ldr->InLoadOrderModuleList.Flink;
 
-	// find loaded module from peb
-	for (auto* dte = (LDR_DATA_TABLE_ENTRY*)ldr->InLoadOrderModuleList.Flink;
-        dte->DllBase != NULL;
-        dte = (LDR_DATA_TABLE_ENTRY*)dte->InLoadOrderLinks.Flink) {
+	// Empty module name implies current process
+	if(m_moduleName.empty()){
+		return reinterpret_cast<uint64_t>(dte->DllBase);

```

Comment:
Please revert all C++ casts to the C style.

Reply:
Sure.